File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if root == None:
            return []
        if (root.left == None and root.right == None):
            return [[root.val]]
        arr = []   # queue
        arr2 = []  # temp array to store level
        res = []
        arr.append(root)
        while len(arr) > 0:
            len_arr = len(arr)
            arr2 = []
            while len_arr > 0:
                node = arr.pop(0)
                arr2.append(node.val)
                if node.left != None:
                    arr.append(node.left)
                if node.right != None:
                    arr.append(node.right)
                len_arr -= 1
            res.append(arr2)
        return res

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> res = new ArrayList<>();
        List<Integer> levelValues;  // list to store current level
        if (root == null) return res;
        
        if (root.left == null && root.right == null) {
            levelValues = new ArrayList<>();
            levelValues.add(root.val);
            res.add(levelValues);
            return res;
        }
        
        Queue<TreeNode> queue = new LinkedList<>();
        TreeNode node;
        int n;
        queue.add(root); 
        while (!queue.isEmpty()) {
            n = queue.size();
            levelValues = new ArrayList<>();
            while (n > 0) {
                node = queue.poll(); 
                levelValues.add(node.val);
                if (node.left != null) {
                    queue.add(node.left);
                }
                if (node.right != null) {
                     queue.add(node.right);
                }
                n--;
            }
            res.add(levelValues);
        }
        return res;
    }
}